internal/cli/dialog/commits_to_beam.go

```diff
@@ -15,15 +15,19 @@ const (
 )
 
 // lets the user select commits to beam to the target branch
-func CommitsToBeam(commits []gitdomain.Commit, targetBranch gitdomain.LocalBranchName, inputs components.TestInput) (gitdomain.Commits, bool, error) {
+func CommitsToBeam(commits []gitdomain.Commit, targetBranch gitdomain.LocalBranchName, formatSHA func(gitdomain.SHA) (string, error), inputs components.TestInput) (gitdomain.Commits, bool, error) {
```

Comment:
can `format` be part of the `SHA` instead of passing it as a function ? like `abbreviated()`, so we can just call `commit.SHA.abbreviated` ?

Reply:
I'm not sure. The difficulty is that it needs to invoke `git`, so it needs a git.Commands and a gitdomain.Runner, or a closure with access to those. I'll wait to see what @kevgo suggests before making changes.

Reply:
```suggestion
func CommitsToBeam(commits []gitdomain.Commit, targetBranch gitdomain.LocalBranchName, git git.Commands, inputs components.TestInput) (gitdomain.Commits, bool, error) {
```

Reply:
I think this would introduce a circular dependency, which is not allowed in Go (for good reasons: to avoid tight coupling). `SHA` exists in the `gitdomain` package. The `git` package depends on the `gitdomain` package. This means we cannot add anything from the `git` package to the `gitdomain` package.

Reply:
Would be nice if we can add this `abbreviatedSha` as part of the `gitdomain.Commit` struct as an `optional` .  Almost all the places where we construct the above struct should have access to the `gitCommands`